samples/extensions/extended_dynamic_state2/extended_dynamic_state2.h

```diff
@@ -0,0 +1,169 @@
+	} ubo_baseline;
+
+	struct UBOTESS
+	{
+		glm::mat4 projection;
+		glm::mat4 modelview;
+		float     tessellation_factor = 1.0f;
+	} ubo_tess;
+
+	struct UBOBG
+	{
+		glm::mat4 projection;
+		glm::mat4 background_modelview;
+
+	} ubo_background;
+
+	VkDescriptorPool descriptor_pool{VK_NULL_HANDLE};
+
+	struct
+	{
+		VkDescriptorSetLayout baseline{VK_NULL_HANDLE};
+		VkDescriptorSetLayout tesselation{VK_NULL_HANDLE};
+		VkDescriptorSetLayout background{VK_NULL_HANDLE};
+	} descriptor_set_layouts;
+
+	struct
+	{
+		VkPipelineLayout baseline{VK_NULL_HANDLE};
+		VkPipelineLayout tesselation{VK_NULL_HANDLE};
+		VkPipelineLayout background{VK_NULL_HANDLE};
+	} pipeline_layouts;
+
+	struct
+	{
+		VkDescriptorSet baseline{VK_NULL_HANDLE};
+		VkDescriptorSet tesselation{VK_NULL_HANDLE};
+		VkDescriptorSet background{VK_NULL_HANDLE};
+	} descriptor_sets;
+
+	struct
+	{
+		VkPipeline baseline{VK_NULL_HANDLE};
+		VkPipeline tesselation{VK_NULL_HANDLE};
+		VkPipeline background{VK_NULL_HANDLE};
+	} pipeline;
+
+	struct
+	{
+		std::unique_ptr<vkb::core::Buffer> baseline;
+		std::unique_ptr<vkb::core::Buffer> tesselation;
+		std::unique_ptr<vkb::core::Buffer> background;
+	} uniform_buffers;
+
+	struct
+	{
+		Texture envmap;
+	} textures;
+
+	struct
+	{
+		glm::mat4 model_matrix;
+		glm::vec4 color;
+	} push_const_block;
+
+	std::unique_ptr<vkb::sg::Scene> scene;
+	struct SceneNode
+	{
+		std::string       name;
+		vkb::sg::Node    *node;
+		vkb::sg::SubMesh *sub_mesh;
+	};
+	std::vector<std::vector<SceneNode>> scene_nodes;
+	typedef enum
+	{
+		SCENE_ALL_OBJ_INDEX = 0,
+		SCENE_BASELINE_OBJ_INDEX,
+		SCENE_TESSELLATION_OBJ_INDEX
+	} sceneObjType_t;
+
+	std::unique_ptr<vkb::sg::SubMesh> background_model;
+
+	struct Cube
+	{
+		std::unique_ptr<vkb::core::Buffer> vertices_pos;
+		std::unique_ptr<vkb::core::Buffer> vertices_norm;
+		std::unique_ptr<vkb::core::Buffer> indices;
+		uint32_t                           index_count;
+	} cube;
+
+	ExtendedDynamicState2();
+	~ExtendedDynamicState2();
+
+	virtual void render(float delta_time) override;
+	virtual void build_command_buffers() override;
+	virtual bool prepare(vkb::Platform &platform) override;
+	virtual void request_gpu_features(vkb::PhysicalDevice &gpu) override;
+	virtual void on_update_ui_overlay(vkb::Drawer &drawer) override;
+	virtual void update(float delta_time) override;
+
+	void prepare_uniform_buffers();
+	void update_uniform_buffers();
+	void create_pipeline();
+	void draw();
+
+	void load_assets();
+	void create_descriptor_pool();
+	void setup_descriptor_set_layout();
+	void create_descriptor_sets();
+	int  get_node_index(std::string name, std::vector<SceneNode> *scene_node);
+	void selection_indicator(const vkb::sg::PBRMaterial *original_mat, vkb::sg::PBRMaterial *new_mat);
+	void scene_pipeline_divide(std::vector<std::vector<SceneNode>> *scene_node);
+	void draw_from_scene(VkCommandBuffer command_buffer, std::vector<std::vector<SceneNode>> *scene_node, sceneObjType_t scene_index);
```

Comment:
Maybe a `const & scene_node` is better suited than a `* scene_node`.

Reply:
fixed